toolium/utils/dataset.py

```diff
@@ -0,0 +1,129 @@
+    if not isinstance(param, string_types):
+        return param
+
+    date_format = "%d/%m/%Y %H:%M:%S" if language == "es" else "%Y/%m/%d %H:%M:%S"
+    date_day_format = "%d/%m/%Y" if language == "es" else "%Y/%m/%d"
+    date_matcher = re.match("\[(NOW|TODAY)\s*([\+|-]\s*\d+)\s*(\w+)\s*\]", param)
+
+    type_mapping_regex = "\[(DICT|LIST|INT|FLOAT|STR):(.*)\]"
+    type_mapping_match_group = re.match(type_mapping_regex, param)
+
+    if "[MISSING_PARAM]" in param:
+        new_param = None
+    elif "[EMPTY]" in param:
+        new_param = param.replace("[EMPTY]", "")
+    elif "[B]" in param:
+        new_param = param.replace("[B]", " ")
+    elif "[RANDOM]" in param:
+        return param.replace("[RANDOM]", uuid.uuid4().hex[0:8])
+    elif "[DATETIME]" in param:
+        return param.replace("[DATETIME]", str(datetime.datetime.utcnow()))
+    elif "[NOW]" in param:
+        return param.replace("[NOW]", str(datetime.datetime.utcnow().strftime(date_format)))
+    elif "[TODAY]" in param:
+        return param.replace("[TODAY]", str(datetime.datetime.utcnow().strftime(date_day_format)))
+    elif "[TRUE]" in param:
+        return True
+    elif "[FALSE]" in param:
+        return False
+    elif "[NULL]" in param:
+        return None
+    elif type_mapping_match_group and type_mapping_match_group.group(1) in \
+            ["LIST", "DICT", "INT", "FLOAT", "STR"]:
+        exec(u"exec_param = {type}({value})".format(type=type_mapping_match_group.group(1).lower(),
+                                                    value=type_mapping_match_group.group(2)))
+        return locals()["exec_param"]
+    elif date_matcher and len(date_matcher.groups()) == 3:
+        configuration = dict([(date_matcher.group(3).lower(), int(date_matcher.group(2).replace(
+            " ", "")))])
+        now = (date_matcher.group(1) == "NOW")
+        reference_date = datetime.datetime.utcnow() if now else datetime.datetime.utcnow().date()
+        replace_value = reference_date + datetime.timedelta(**configuration)
+        return replace_value.strftime(date_format) if now else replace_value.strftime(
+            date_day_format)
+    else:
+        new_param = generate_fixed_length_param(param)
+    logger.debug("Input param: %s, output param: %s" % (param, new_param))
+    return new_param
+
+
+def generate_fixed_length_param(param):
+    """
+    Generate a fixed length param if the elements matches the expression
+    [<type>_WITH_LENGTH_<length>] where <type> can be:
+    STRING, INTEGER, STRING_ARRAY, INTEGER_ARRAY, JSON.
+    E.g.: [STRING_WITH_LENGTH_15]
+    :param param: Lettuce param
+    :return param with the desired length
+    """
+    if param.startswith("[") and param.endswith("]"):
+        if "_ARRAY_WITH_LENGTH_" in param:
+            seeds = {"STRING": "a", "INTEGER": 1}
+            seed, length = param[1:-1].split("_ARRAY_WITH_LENGTH_")
+            param = list(seeds[seed] for x in xrange(int(length)))
```

Comment:
it will fail if seed is different from STRING or INTEGER, may be `if "_ARRAY_WITH_LENGTH_" in param:` should be check if it is a valid param value